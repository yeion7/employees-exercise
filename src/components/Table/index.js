import React from 'react'
import PropTypes from 'prop-types'
import Row from '../Row'

const Heading = ({text, name, sortBy, sortedBy}) => {
  return (<th
    onClick={sortBy(name)}
    style={{background: `${sortedBy === name ? 'palegreen' : ''}`}}>
    {text}
  </th>)
}

const Table = ({employees, sortBy, sortedBy}) => {
  return (
    <table>
      <thead>
        <tr>
          <Heading sortBy={sortBy} name='name' text='nombre' sortedBy={sortedBy} />
          <Heading sortBy={sortBy} name='company' text='Compañía' sortedBy={sortedBy} />
          <Heading sortBy={sortBy} name='salary' text='Salario' sortedBy={sortedBy} />
          <Heading sortBy={sortBy} name='age' text='Edad' sortedBy={sortedBy} />
          <Heading sortBy={sortBy} name='phone' text='Télefono' sortedBy={sortedBy} />
          <Heading sortBy={sortBy} name='email' text='Correo' sortedBy={sortedBy} />
          <th>Borrar</th>
        </tr>
      </thead>
      <tbody>
        {
          employees.map(employ => <Row key={employ.id} {...employ} />)
        }
      </tbody>
    </table>
  )
}

Table.propTypes = {
  employees: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      company: PropTypes.string.isRequired,
      salary: PropTypes.number.isRequired,
      age: PropTypes.number.isRequired,
      phone: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired
    })
  ).isRequired
}

export default Table
