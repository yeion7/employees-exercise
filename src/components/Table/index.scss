.table {
  // display: block;
  overflow-x: auto;
  white-space: nowrap;
  min-width: 100%;

  &__head > tr {
    background: #f4b642;
    font-weight: bold;
    cursor: pointer;
  }
}

.heading {
  &__cell {
    padding: 10px 0;
  }
}
