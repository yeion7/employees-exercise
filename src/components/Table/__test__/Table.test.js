import React from 'react'
import renderer from 'react-test-renderer'

import Table from '../'
import employees from '../__mocks__/employees'

describe('Table component', () => {
  it('renders correctly', () => {
    const fn = jest.fn()
    const factory = jest.fn(() => jest.fn())

    const tree = renderer
      .create(
        <Table
          employees={employees}
          sortBy={factory}
          sortedBy=""
          currency="MXN"
          changeType="21.9"
          deleteEmploy={fn}
          search=""
          onEditField={factory}
          editable={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with search', () => {
    const fn = jest.fn()
    const factory = jest.fn(() => jest.fn())

    const tree = renderer
      .create(
        <Table
          employees={employees}
          sortBy={factory}
          sortedBy=""
          currency="MXN"
          changeType="21.9"
          deleteEmploy={fn}
          search="Miguel Ángel Rodríguez"
          onEditField={factory}
          editable={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with sorter and USD', () => {
    const fn = jest.fn()
    const factory = jest.fn(() => jest.fn())

    const tree = renderer
      .create(
        <Table
          employees={employees}
          sortBy={factory}
          sortedBy="age"
          currency="USD"
          changeType="21.9"
          deleteEmploy={fn}
          search=""
          onEditField={factory}
          editable={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with editable mode', () => {
    const fn = jest.fn()
    const factory = jest.fn(() => jest.fn())

    const tree = renderer
      .create(
        <Table
          employees={employees}
          sortBy={factory}
          sortedBy="age"
          currency="USD"
          changeType="21.9"
          deleteEmploy={fn}
          search=""
          onEditField={factory}
          editable={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
