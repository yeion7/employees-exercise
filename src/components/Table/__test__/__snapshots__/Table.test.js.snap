// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th
        data-test="sort-name"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        nombre
      </th>
      <th
        data-test="sort-company"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Compañía
      </th>
      <th
        data-test="sort-salary"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Salario
      </th>
      <th
        data-test="sort-age"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Edad
      </th>
      <th
        data-test="sort-phone"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Télefono
      </th>
      <th
        data-test="sort-email"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Correo
      </th>
      <th>
        Borrar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span>
          Eduardo González
        </span>
      </td>
      <td>
        <span>
          América Móvil
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "indianred",
            }
          }
        >
          MX$ 5,400.00
        </span>
      </td>
      <td>
        <span>
          57
        </span>
      </td>
      <td>
        <span>
          (886) 271 4211
        </span>
      </td>
      <td>
        <span>
          eduardo@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-0"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Miguel Ángel Rodríguez
        </span>
      </td>
      <td>
        <span>
          FEMSA
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 11,000.00
        </span>
      </td>
      <td>
        <span>
          40
        </span>
      </td>
      <td>
        <span>
          (197) 651 8811
        </span>
      </td>
      <td>
        <span>
          miguel@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-1"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Mariana Gómez
        </span>
      </td>
      <td>
        <span>
          Banorte
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 16,900.00
        </span>
      </td>
      <td>
        <span>
          34
        </span>
      </td>
      <td>
        <span>
          (396) 702 3554
        </span>
      </td>
      <td>
        <span>
          mariana@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-2"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Alejandro Fernández
        </span>
      </td>
      <td>
        <span>
          Grupo México
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 17,700.00
        </span>
      </td>
      <td>
        <span>
          41
        </span>
      </td>
      <td>
        <span>
          (542) 591 3978
        </span>
      </td>
      <td>
        <span>
          alejandro@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-3"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Gabriela López
        </span>
      </td>
      <td>
        <span>
          Grupo Inbursa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 10,100.00
        </span>
      </td>
      <td>
        <span>
          65
        </span>
      </td>
      <td>
        <span>
          (783) 560 4847
        </span>
      </td>
      <td>
        <span>
          gabriela@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-4"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Alejandra Díaz
        </span>
      </td>
      <td>
        <span>
          Cemex
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "indianred",
            }
          }
        >
          MX$ 6,500.00
        </span>
      </td>
      <td>
        <span>
          27
        </span>
      </td>
      <td>
        <span>
          (862) 721 8605
        </span>
      </td>
      <td>
        <span>
          alejandra@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-5"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Juan Carlos Martínez
        </span>
      </td>
      <td>
        <span>
          Grupo Televisa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 14,000.00
        </span>
      </td>
      <td>
        <span>
          38
        </span>
      </td>
      <td>
        <span>
          (973) 322 3993
        </span>
      </td>
      <td>
        <span>
          juan@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-6"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          José Luis Pérez
        </span>
      </td>
      <td>
        <span>
          Alfa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 20,400.00
        </span>
      </td>
      <td>
        <span>
          50
        </span>
      </td>
      <td>
        <span>
          (226) 806 3427
        </span>
      </td>
      <td>
        <span>
          jose@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-7"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          María Carmen García
        </span>
      </td>
      <td>
        <span>
          Grupo Bimbo
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 23,700.00
        </span>
      </td>
      <td>
        <span>
          29
        </span>
      </td>
      <td>
        <span>
          (907) 702 7639
        </span>
      </td>
      <td>
        <span>
          maria@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-8"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Adriana Sánchez
        </span>
      </td>
      <td>
        <span>
          Liverpool
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 33,800.00
        </span>
      </td>
      <td>
        <span>
          23
        </span>
      </td>
      <td>
        <span>
          (248) 255 7702
        </span>
      </td>
      <td>
        <span>
          adriana@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-9"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component renders correctly with editable mode 1`] = `
<table>
  <thead>
    <tr>
      <th
        data-test="sort-name"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        nombre
      </th>
      <th
        data-test="sort-company"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Compañía
      </th>
      <th
        data-test="sort-salary"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Salario
      </th>
      <th
        data-test="sort-age"
        onClick={[Function]}
        style={
          Object {
            "background": "palegreen",
          }
        }
      >
        Edad
      </th>
      <th
        data-test="sort-phone"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Télefono
      </th>
      <th
        data-test="sort-email"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Correo
      </th>
      <th>
        Borrar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <input
          data-test="input-name-0"
          defaultValue="Eduardo González"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          América Móvil
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-0"
          defaultValue={5400}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-0"
          defaultValue={57}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-0"
          defaultValue="(886) 271 4211"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-0"
          defaultValue="eduardo@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-0"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-1"
          defaultValue="Miguel Ángel Rodríguez"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          FEMSA
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-1"
          defaultValue={11000}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-1"
          defaultValue={40}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-1"
          defaultValue="(197) 651 8811"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-1"
          defaultValue="miguel@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-1"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-2"
          defaultValue="Mariana Gómez"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Banorte
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-2"
          defaultValue={16900}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-2"
          defaultValue={34}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-2"
          defaultValue="(396) 702 3554"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-2"
          defaultValue="mariana@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-2"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-3"
          defaultValue="Alejandro Fernández"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Grupo México
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-3"
          defaultValue={17700}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-3"
          defaultValue={41}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-3"
          defaultValue="(542) 591 3978"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-3"
          defaultValue="alejandro@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-3"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-4"
          defaultValue="Gabriela López"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Grupo Inbursa
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-4"
          defaultValue={10100}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-4"
          defaultValue={65}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-4"
          defaultValue="(783) 560 4847"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-4"
          defaultValue="gabriela@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-4"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-5"
          defaultValue="Alejandra Díaz"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Cemex
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-5"
          defaultValue={6500}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-5"
          defaultValue={27}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-5"
          defaultValue="(862) 721 8605"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-5"
          defaultValue="alejandra@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-5"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-6"
          defaultValue="Juan Carlos Martínez"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Grupo Televisa
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-6"
          defaultValue={14000}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-6"
          defaultValue={38}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-6"
          defaultValue="(973) 322 3993"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-6"
          defaultValue="juan@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-6"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-7"
          defaultValue="José Luis Pérez"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Alfa
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-7"
          defaultValue={20400}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-7"
          defaultValue={50}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-7"
          defaultValue="(226) 806 3427"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-7"
          defaultValue="jose@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-7"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-8"
          defaultValue="María Carmen García"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Grupo Bimbo
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-8"
          defaultValue={23700}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-8"
          defaultValue={29}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-8"
          defaultValue="(907) 702 7639"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-8"
          defaultValue="maria@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-8"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <input
          data-test="input-name-9"
          defaultValue="Adriana Sánchez"
          onChange={[Function]}
        />
      </td>
      <td>
        <span>
          Liverpool
        </span>
      </td>
      <td>
        <input
          data-test="input-salary-9"
          defaultValue={33800}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-age-9"
          defaultValue={23}
          onChange={[Function]}
          type="number"
        />
      </td>
      <td>
        <input
          data-test="input-phone-9"
          defaultValue="(248) 255 7702"
          onChange={[Function]}
        />
      </td>
      <td>
        <input
          data-test="input-email-9"
          defaultValue="adriana@gmail.com"
          onChange={[Function]}
          type="email"
        />
      </td>
      <td>
        <button
          data-test="delete-9"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component renders correctly with search 1`] = `
<table>
  <thead>
    <tr>
      <th
        data-test="sort-name"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        nombre
      </th>
      <th
        data-test="sort-company"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Compañía
      </th>
      <th
        data-test="sort-salary"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Salario
      </th>
      <th
        data-test="sort-age"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Edad
      </th>
      <th
        data-test="sort-phone"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Télefono
      </th>
      <th
        data-test="sort-email"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Correo
      </th>
      <th>
        Borrar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span>
          Miguel Ángel Rodríguez
        </span>
      </td>
      <td>
        <span>
          FEMSA
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          MX$ 11,000.00
        </span>
      </td>
      <td>
        <span>
          40
        </span>
      </td>
      <td>
        <span>
          (197) 651 8811
        </span>
      </td>
      <td>
        <span>
          miguel@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-1"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table component renders correctly with sorter and USD 1`] = `
<table>
  <thead>
    <tr>
      <th
        data-test="sort-name"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        nombre
      </th>
      <th
        data-test="sort-company"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Compañía
      </th>
      <th
        data-test="sort-salary"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Salario
      </th>
      <th
        data-test="sort-age"
        onClick={[Function]}
        style={
          Object {
            "background": "palegreen",
          }
        }
      >
        Edad
      </th>
      <th
        data-test="sort-phone"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Télefono
      </th>
      <th
        data-test="sort-email"
        onClick={[Function]}
        style={
          Object {
            "background": "",
          }
        }
      >
        Correo
      </th>
      <th>
        Borrar
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span>
          Eduardo González
        </span>
      </td>
      <td>
        <span>
          América Móvil
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "indianred",
            }
          }
        >
          US$ 246.00
        </span>
      </td>
      <td>
        <span>
          57
        </span>
      </td>
      <td>
        <span>
          (886) 271 4211
        </span>
      </td>
      <td>
        <span>
          eduardo@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-0"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Miguel Ángel Rodríguez
        </span>
      </td>
      <td>
        <span>
          FEMSA
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 502.00
        </span>
      </td>
      <td>
        <span>
          40
        </span>
      </td>
      <td>
        <span>
          (197) 651 8811
        </span>
      </td>
      <td>
        <span>
          miguel@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-1"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Mariana Gómez
        </span>
      </td>
      <td>
        <span>
          Banorte
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 771.00
        </span>
      </td>
      <td>
        <span>
          34
        </span>
      </td>
      <td>
        <span>
          (396) 702 3554
        </span>
      </td>
      <td>
        <span>
          mariana@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-2"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Alejandro Fernández
        </span>
      </td>
      <td>
        <span>
          Grupo México
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 808.00
        </span>
      </td>
      <td>
        <span>
          41
        </span>
      </td>
      <td>
        <span>
          (542) 591 3978
        </span>
      </td>
      <td>
        <span>
          alejandro@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-3"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Gabriela López
        </span>
      </td>
      <td>
        <span>
          Grupo Inbursa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 461.00
        </span>
      </td>
      <td>
        <span>
          65
        </span>
      </td>
      <td>
        <span>
          (783) 560 4847
        </span>
      </td>
      <td>
        <span>
          gabriela@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-4"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Alejandra Díaz
        </span>
      </td>
      <td>
        <span>
          Cemex
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "indianred",
            }
          }
        >
          US$ 296.00
        </span>
      </td>
      <td>
        <span>
          27
        </span>
      </td>
      <td>
        <span>
          (862) 721 8605
        </span>
      </td>
      <td>
        <span>
          alejandra@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-5"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Juan Carlos Martínez
        </span>
      </td>
      <td>
        <span>
          Grupo Televisa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 639.00
        </span>
      </td>
      <td>
        <span>
          38
        </span>
      </td>
      <td>
        <span>
          (973) 322 3993
        </span>
      </td>
      <td>
        <span>
          juan@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-6"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          José Luis Pérez
        </span>
      </td>
      <td>
        <span>
          Alfa
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 931.00
        </span>
      </td>
      <td>
        <span>
          50
        </span>
      </td>
      <td>
        <span>
          (226) 806 3427
        </span>
      </td>
      <td>
        <span>
          jose@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-7"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          María Carmen García
        </span>
      </td>
      <td>
        <span>
          Grupo Bimbo
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 1,082.00
        </span>
      </td>
      <td>
        <span>
          29
        </span>
      </td>
      <td>
        <span>
          (907) 702 7639
        </span>
      </td>
      <td>
        <span>
          maria@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-8"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <span>
          Adriana Sánchez
        </span>
      </td>
      <td>
        <span>
          Liverpool
        </span>
      </td>
      <td>
        <span
          style={
            Object {
              "color": "seagreen",
            }
          }
        >
          US$ 1,543.00
        </span>
      </td>
      <td>
        <span>
          23
        </span>
      </td>
      <td>
        <span>
          (248) 255 7702
        </span>
      </td>
      <td>
        <span>
          adriana@gmail.com
        </span>
      </td>
      <td>
        <button
          data-test="delete-9"
          onClick={[Function]}
        >
          Borrar
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
